src/lib/containers/QueryWrapper.tsx

```diff
@@ -340,7 +347,7 @@ export function QueryWrapper(props: QueryWrapperProps) {
     entity,
     executeQueryRequest,
     isFacetsAvailable,
-    asyncJobStatus: infiniteData?.pages[0],
+    asyncJobStatus: currentAsyncStatus,
```

Comment:
This didn't work because `infiniteData?.pages[0]` was undefined until the request for the first page completed. The "PROCESSING" state was never reported.

Reply:
So you added a callback in the SynapseClient function, looks good to me